sky/utils/command_runner.py

```diff
@@ -84,17 +90,32 @@ def ssh_options_list(ssh_private_key: Optional[str],
         ssh_private_key,
     ] if ssh_private_key is not None else []
 
+    arg_dict = {k: [v] for k, v in arg_dict.items()}
+    arg_dict = collections.defaultdict(list, arg_dict)
+
     if ssh_proxy_command is not None:
         logger.debug(f'--- Proxy: {ssh_proxy_command} ---')
-        arg_dict.update({
-            # Due to how log_lib.run_with_log() works (using shell=True) we
-            # must quote this value.
-            'ProxyCommand': shlex.quote(ssh_proxy_command),
-        })
+        # Due to how log_lib.run_with_log() works (using shell=True) we
+        # must quote this value.
+        arg_dict['ProxyCommand'].append(shlex.quote(ssh_proxy_command))
+
+    if docker_ssh_proxy_command is not None:
+        logger.debug(f'--- Docker SSH Proxy: {docker_ssh_proxy_command} ---')
+        # If two proxy commands are specified, e.g.
+        # ssh -o ProxyCommand=cmd1 -o ProxyCommand=cmd2 ...
+        # then ssh will first ssh into the host using cmd2 (the later one) and
+        # then ssh from cmd2 host to the target host using cmd1. So, we put the
+        # docker proxy command at the beginning. This way, ssh will first
+        # establish a connection to the user-specified proxy host and then
+        # proceed to ssh from the proxy host to the docker host VM, and
+        # finally, ssh into the docker container.
+        arg_dict['ProxyCommand'].insert(0,
+                                        shlex.quote(docker_ssh_proxy_command))
```

Comment:
nit: if we place this `if` block before the `if ssh_proxy_command is not None`, we can use `append` instead of `insert`?